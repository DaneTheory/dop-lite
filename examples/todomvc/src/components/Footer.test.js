import React from 'react'
import renderer from 'react-test-renderer';
import Footer from './Footer'
import { addTodo, completeTodo } from '../actions'

let component
test('Checking item lefts', () => {
  component = renderer.create(
    <Footer />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // No items left
  expect(tree.children[0].children[2]).toBe('items')
  expect(tree.children[0].children[0].children[0]).toBe('No')

  // Adding todo manually
  const labelnewtodo = 'My new todo'
  addTodo(labelnewtodo)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // 1 item left
  expect(tree.children[0].children[2]).toBe('item')
  expect(tree.children[0].children[0].children[0]).toBe(1)

  // Adding todo manually
  addTodo('My new todo 2')
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // 2 item left
  expect(tree.children[0].children[2]).toBe('items')
  expect(tree.children[0].children[0].children[0]).toBe(2)

  // Completing todo manually
  completeTodo(1)
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // 1 item left
  expect(tree.children[0].children[2]).toBe('item')
  expect(tree.children[0].children[0].children[0]).toBe(1)
});

