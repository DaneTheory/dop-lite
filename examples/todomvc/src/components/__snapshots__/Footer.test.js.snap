// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking item lefts 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      No
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className="selected"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
</footer>
`;

exports[`Checking item lefts 2`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      1
    </strong>
     
    item
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className="selected"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
</footer>
`;

exports[`Checking item lefts 3`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      2
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className="selected"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
</footer>
`;

exports[`Checking item lefts 4`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      1
    </strong>
     
    item
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className="selected"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;
